core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/logprovider/buffer_v1.go

```diff
@@ -379,7 +422,13 @@ func (q *upkeepLogQueue) enqueue(blockThreshold int64, logsToAdd ...logpoller.Lo
 		}
 		q.states[lid] = logTriggerStateEntry{state: logTriggerStateEnqueued, block: log.BlockNumber}
 		added++
-		logs = append(logs, log)
+		if logList, ok := q.logs[log.BlockNumber]; ok {
+			logList = append(logList, log)
+			q.logs[log.BlockNumber] = logList
+		} else {
+			q.logs[log.BlockNumber] = []logpoller.Log{log}
+			q.blockNumbers = append(q.blockNumbers, log.BlockNumber)
+		}
```

Comment:
we are setting values on q.logs here but we override q.logs in line 433. can we verify tests for this function and ensure behaviour is expected with line 433

Reply:
This has been fixed; we no longer set the value of q.logs